docs/future_plans.md

```diff
@@ -0,0 +1,62 @@
+**Last updated:** January 2021.
+
+**Tl;dr:** We'll be redesigning how Stardoc works, and deprioritizing feature
+requests and minor bugs until that work is complete (targeting 2021 Q2).
+
+## Technical motivation
+
+Stardoc is currently the recommended tool for generating documentation of
+Starlark rules. It [replaces](skydoc_deprecation.md) Skydoc, the previous tool,
+which worked by mocking evaluation of .bzl files as Python code.
+
+Mocking is an inherently problematic approach for two reasons:
+
+1. It creates a maintenance burden for the tool maintainer (us). We have to
+ensure that the mocked definitions stay up-to-date as Bazel changes. These
+include not just `rule()` and `provider()`, but also a number of other symbols
+that don't directly affect documentation but still require stubs.
+
+2. It puts a constraint on the user: All of their documented .bzl files, as well
+as all of the .bzl dependencies they transitively load, must be compatible with
+the mock evaluation. This means users must be vigilant about writing Starlark
+code that lies in the intersection of what is understood as valid by Bazel and
+by Stardoc.
+
+Skydoc experienced an extreme version of this problem because it didn't even
+treat .bzl files as being written in the Starlark language. However, Stardoc
+also uses mocking -- not of the Starlark language, but of Bazel's built-in
+symbols for writing rules (the Build Language).
+
+In addition, Stardoc's mocking approach tightly integrates it with Bazel. Indeed
+much of its source code lives inside the bazelbuild/bazel repository. This makes
+refactoring and evolving the Bazel source code more difficult.
+
+## Our plans
+
+We will rewrite the part of Stardoc that extracts documentation information from
+.bzl files, so that instead of using mocking to pseudo-evaluate individual .bzl
+files, it performs a real Bazel evaluation of the workspace. Think of how `bazel
+query` is used to dump out information from Bazel's loading phase about the
```

Comment:
Perhaps add words to this effect, or parts thereof:

Bazel's build Language is significantly more complex than core Starlark, and though there are several implementations of Starlark, there is only one tool capable of accurately interpreting BUILD and .bzl files: Bazel. Designs for new tools should not fall into the same trap of assuming that the build language is simple enough to be accurately simulated; instead, they should ask Bazel for the correct information, and Bazel should be extended to answer all reasonable queries about package loading concepts.

Reply:
Added a paragraph above. I stopped short of recommending querying Bazel as a one-size-fits-all solution. For instance, I can see why tools that statically edit BUILD files may not want to have a running Bazel process.